chutney/server/src/test/java/com/chutneytesting/execution/infra/storage/DatabaseExecutionHistoryRepositoryTest.java

```diff
@@ -463,6 +464,71 @@ void deletes_executions_by_ids() {
             ).isInstanceOf(ReportNotFoundException.class));
         }
 
+        @Test
+        void campaign_execution_must_be_deleted_when_last_scenario_execution_is_deleted() {
+            // GIVEN
+            ScenarioEntity scenarioEntity = givenScenario();
+            CampaignEntity campaign = givenCampaign(scenarioEntity);
+            ScenarioExecutionEntity scenarioExecutionOne = givenScenarioExecution(scenarioEntity.getId(), FAILURE);
+            ScenarioExecutionCampaign scenarioExecutionOneReport = new ScenarioExecutionCampaign(scenarioEntity.getId().toString(), scenarioEntity.getTitle(), scenarioExecutionOne.toDomain());
+
+            Long campaignExecutionId = campaignExecutionDBRepository.generateCampaignExecutionId(campaign.id(), "executionEnv");
+            CampaignExecution campaignExecution = CampaignExecutionReportBuilder.builder()
+                .executionId(campaignExecutionId)
+                .campaignId(campaign.id())
+                .campaignName(campaign.title())
+                .partialExecution(true)
+                .environment("env")
```

Comment:
Not the same environment as line 475 !! If it's not important then try null or empty value, if not possible then why not reuse the environment from th scenario execution ?